src/vcStringFormat.cpp

```diff
@@ -26,25 +25,27 @@ const char *vcStringFormat(const char *pFormatString, const char **ppStrings, si
 
       int length;
       currArg = udStrAtoi((pFormatString + i + 1), &length);
-      if (currArg > (int) numStrings || currArg < 0 || pFormatString[length + i + 1] != '}')
+      if (currArg > numStrings || pFormatString[length + i + 1] != '}')
         continue;
 
-      newChars += (int) udStrlen(ppStrings[currArg]);
+      newChars += (int)udStrlen(ppStrings[currArg]);
       newChars -= length + 2;
       i += length - 1;
     }
   }
 
-  len = i;
+  if (pOldLength != nullptr)
+    *pOldLength = i;
 
-  if (len == 0)
-    return pFormatString;
+  return i + newChars + 1;
+}
 
-  int newLength = len + newChars + 1;
-  char *pBuf = udAllocType(char, newLength, udAF_None);
+const char *vcStringFormat_FillBuffer(char *pBuf, const char *pFormatString, const char **ppStrings, size_t numStrings)
```

Comment:
This should really take a buffer length, to avoid overrunning the buffer. It also seems unnecessary to have this function return `pBuf`?

Reply:
In every instance of its use it's called after constructing a buffer of the perfect size or checking the length of the input buffer, this isn't intended for use outside of vcStringFormat.cpp

Reply:
I'm not really worried about the usage otuside of this file - if it's not intented to be used outside of the file, you should mark them as `static` though. I'm far more worried about the two functions ending up out of sync with each other, data corruption is not fun to track down and I'd rather avoid it entirely.

Reply:
I've refactored the buffer argument version of the function to no longer calculate output size beforehand, in this iteration it will just output as much as it can and report how many characters were written, making this no longer a concern